src/wodan/wodan_lru.ml

```diff
@@ -0,0 +1,78 @@
+module type LRUValue = sig
+  type t
+
+  val discardable : t -> bool
+end
+
+module Make (K : Hashtbl.HashedType) (V : LRUValue) = struct
+  module Val = struct
+    include V
+
+    let weight _ = 1
+  end
+
+  module LRU = Lru.M.Make (K) (Val)
+
+  type t = LRU.t
+
+  type key = K.t
+
+  type value = V.t
+
+  exception Already_cached of key
+
+  exception Too_small
+
+  let get_opt t k = LRU.find ~promote:true k t
+
+  let get t k =
+    match get_opt t k with
+    | None ->
+        raise Not_found
+    | Some x ->
+        x
```

Comment:
This implementation seems like the worst of all worlds, since it allocates an option value and then turns it into a fast exception :-)  Wouldn't it be better to use the convention of having `get` functions return an option type and expose ` get_exn` that raises an exception?

Reply:
an other convention is to use `get` for exceptions throwing functions and `find` for the ones returning an option.

Reply:
Obviously true about the implementation... :thinking: 
About the naming, I used the `stdlib` convention, but I don't mind changing it to the other standard, as long as we keep it consistent.

Reply:
I looked into it, actually this was already happening before in `wodan.ml` ; we had many pattern matching that were all raising the same exception `MissingLRUEntry` and created a lot of (IMO) useless nested code that could be flattened by a raising exception function.
Besides, `lru` does not provide a `find_exn`, this is why I did it by unwrapping the option.
Also note that this exception (as well as many others) should not be raised during normal use because it reflects a LRU invariant, this is why it was not (and is not) caught, but I agree we should probably have a better handling in the future.
Anyway I renamed it to match the former version, and avoid out-of-the-blue `Not_found` exceptions.

Please let me know if you still think this is not reasonable.